posydon/popsyn/rate_calculation.py

```diff
@@ -689,20 +703,230 @@ def compute_merger_rate_density(self, w_ijk, z_event, observable='DCOs', sensiti
         z_hor = self.get_edges_redshift_bins()
         n = len(z_hor)
 
-        if observable=='DCOs':
+        if observable=='DCO':
             z_merger_DCO = z_event
-            Rate_DCOs = np.zeros(n-1)
+            Rate_DCO = np.zeros(n-1)
             if sensitivity=='infinite':
                 for i in range(1,n):
                     # compute Eq. (D.1) in Bavera et al. (2022) arXiv:2106.15841
-                    condition_DCOs = np.logical_and(z_merger_DCO>z_hor[i-1],
+                    cond_DCO = np.logical_and(z_merger_DCO>z_hor[i-1],
                                                     z_merger_DCO<=z_hor[i])
-                    Rate_DCOs[i-1] = (sum(w_ijk[condition_DCOs])
+                    Rate_DCO[i-1] = (sum(w_ijk[cond_DCO])
                                      /self.get_shell_comovig_volume(z_hor[i-1],
                                                                     z_hor[i],
                                                                     sensitivity))
-                return Rate_DCOs # Gpc^-3 yr^-1
+                return Rate_DCO # Gpc^-3 yr^-1
             else:
                 raise ValueError('Unsupported sensitivity!')
+        
+        elif observable in ['GRB1','GRB2']:
+            s = observable[-1]
```

Comment:
It is unclear to me from here and the raise ValueError that the user will know observable should be one string

Reply:
I now have updated the docstring accordingly. Note that this method will not be accessible by the user, the user will use the `SyntheticPopulation` method `get_grb_rate_density` that will take care of computing the GRB rate density from `star_1` and `star_2` and return the corresponding rate densities for both stars and total, as well as the intrinsic and observable data frames (similar to the DCO population).

Reply:
Here I'd like to point you to the document Konstantinos pinned in the developers slack channel:
In section 6 it recommends to start all private functions in a class with an underscore `_` to show, that this function should not be called from outside.

Reply:
Correct, unfortunately, we never enforced this condition. Something I plan to do, we could do it together over the summer, is to enforce a shared Python formatting and syntax and have a GitHub action rejecting any unformatted PR. Then we will request everyone to set up the code editor to run the formatted on save for each file. I was playing around with the VS Code Python formatted and tested the `yapf` with `pep8` but is not matching the v1 syntax. I am waiting for @kkovlakas to get back to me with the exact setting used to format the v1 code. Once this is set we parse all files to set the proper formatting and eventually homogenize class and method names.